### PR TITLE
PLT-2992 Added the ability to use different themes for each team

### PR DESCRIPTION
**Pull request by hmhealey**
This patch moves the user's ThemeProps field into the preferences table. Each preferences entry has the category set to "theme" and name set to either a team id or the empty string. If there is a preferences entry for a given team, it'll use that one, but if not, it'll use the default one with the empty string. When the user sets a new default, it erases all of the team specific ones.

Also does the following to support this:
1) Adds a delete preference route which takes a list of preference objects, similar to the save preference route (/api/v3/preferences/delete)
2) Increases the size of the Preferences.Value field to 2000 to match the length of the User.ThemeProps field
3) Breaks up utils/constants.jsx so that it can be imported as before with `import Constants from 'utils/constants.jsx';` or with some of the individual parts using `import {ActionTypes, Preferences} from 'utils/constants.jsx';` (https://mattermost.atlassian.net/browse/PLT-2646)

### CODE REVIEW

**Comment by lfbrock** [comment]
Tested: 

1. Changing theme when you only belong only one team
2. Changing themes when you belong to two teams
  - Changing theme (without applying to all), and switching teams
  - Changing theme (applying to all), and switching teams

Bugs:

1. With only one team created, choose a default theme option (eg "Mattermost") and press Save. The theme does not save, and reverts back to the original theme. (Note: Custom theme seems to save fine, and you can save a default theme after you save it once using the "apply to all" checked.) 

Doc update: https://github.com/mattermost/docs/pull/263

**Comment by hmhealey** [comment]
@lfbrock That issue should be fixed

**Comment by hmhealey** [comment]
I'll create a ticket to look into how we can improve that since all settings use the same set of save/cancel buttons so this sort of thing may affect all of them. They should really follow our UX guidelines

File: webapp/utils/async_client.jsx [hunk 1]
```diff
@@ -808,6 +808,29 @@ export function savePreferences(preferences, success, error) {
     );
 }
 
+export function deletePreferences(preferences, success, error) {
```
Comment (jwilander) [inline webapp/utils/async_client.jsx]: Why the wrapper? Why not just use the Client directly in actions?

Comment (hmhealey) [inline webapp/utils/async_client.jsx]: I wanted to stay consistent with other calls where there's an async version and a regular one. I imagine that there's other pieces of code that might want to call this in the future without talking to the client directly.

**Comment by coreyhulen** [comment]
+1

**Comment by enahum** [comment]
needs rebase

**Pull request merged**